tests/components/dexcom/test_sensor.py

```diff
@@ -0,0 +1,114 @@
+"""The sensor tests for the griddy platform."""
+
+from pydexcom import SessionError
+
+from homeassistant.components.dexcom.const import MMOL_L
+from homeassistant.const import (
+    CONF_UNIT_OF_MEASUREMENT,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+
+from tests.async_mock import patch
+from tests.components.dexcom import GLUCOSE_READING, init_integration
+
+
+async def test_sensors(hass):
+    """Test we get sensor data."""
+    await init_integration(hass)
+
+    test_username_glucose_value = hass.states.get(
+        "sensor.dexcom_test_username_glucose_value"
+    )
+    assert test_username_glucose_value.state == str(GLUCOSE_READING.value)
+    test_username_glucose_trend = hass.states.get(
+        "sensor.dexcom_test_username_glucose_trend"
+    )
+    assert test_username_glucose_trend.state == GLUCOSE_READING.trend_description
+
+
+async def test_sensors_unknown(hass):
+    """Test we handle sensor state unknown."""
+    await init_integration(hass)
+
+    with patch(
+        "homeassistant.components.dexcom.Dexcom.get_current_glucose_reading",
+        return_value=None,
+    ):
+        await hass.helpers.entity_component.async_update_entity(
+            "sensor.dexcom_test_username_glucose_value"
+        )
+        await hass.helpers.entity_component.async_update_entity(
+            "sensor.dexcom_test_username_glucose_trend"
+        )
+
+    test_username_glucose_value = hass.states.get(
+        "sensor.dexcom_test_username_glucose_value"
+    )
+    assert test_username_glucose_value.state == STATE_UNKNOWN
+    test_username_glucose_trend = hass.states.get(
+        "sensor.dexcom_test_username_glucose_trend"
+    )
+    assert test_username_glucose_trend.state == STATE_UNKNOWN
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)